### PR TITLE
Move to Kotlin 1.9.23 and clean up build and source files

### PR DESCRIPTION
**Pull request by findjigar**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -55,73 +47,30 @@ kotlin {
...
+        implementation(kotlin("test"))
     }
 }
 
-apply(plugin = "com.vanniktech.maven.publish")
+// https://github.com/Kotlin/kotlin-wasm-examples/commit/701a051d6ee869abcabebff702b3ccd98d51c38d
+rootProject.the<NodeJsRootExtension>().apply {
+    nodeVersion = "21.0.0-v8-canary202309143a48826a08"
+    nodeDownloadBaseUrl = "https://nodejs.org/download/v8-canary"
+}
```
Comment (faogustavo) [inline build.gradle.kts:76]: Do you know if this is required for libraries? I thought they were only needed for the module that would produce the final output.

File: build.gradle.kts [hunk 2]
```diff
@@ -55,73 +47,30 @@ kotlin {
         browser()
         nodejs()
     }
-    wasm {
+    @OptIn(ExperimentalWasmDsl::class)
+    wasmJs {
         browser()
+        nodejs()
         binaries.executable()
     }
```
Comment (faogustavo) [inline build.gradle.kts:55]: I think we can simplify as in the example below. What do you think? AFAIK These configurations are only for the final output.

```suggestion
    wasmJs()
```

Comment (faogustavo) [reply build.gradle.kts:55]: If we can remove, also remove on JS please.

File: build.gradle.kts (hunk 1 above)
Comment (findjigar) [reply build.gradle.kts:76]: I think I needed this for running tests. For some reason, they were failing. I'll double check

File: build.gradle.kts (hunk 2 above)
Comment (findjigar) [reply build.gradle.kts:55]: What do you mean when you say `These configurations are only for the final output.` ?

Comment (findjigar) [reply build.gradle.kts:55]: Ktor is a library and it defines the targets
https://github.com/ktorio/ktor/blob/6e52d76177a823b3c667989483e9bce35b04f53c/buildSrc/src/main/kotlin/TargetsConfig.kt#L45

File: build.gradle.kts (hunk 1 above)
Comment (findjigar) [reply build.gradle.kts:76]: Ktor has it and also had a comment related to why it's there
https://github.com/ktorio/ktor/blob/6e52d76177a823b3c667989483e9bce35b04f53c/build.gradle.kts#L229

I'll still check if the project works without it. In that case, I'll remove it

Comment (findjigar) [reply build.gradle.kts:76]: My recollection was right. Test failure without the above block
Screenshot 2024-03-20 at 11 46 18 PM

**Pull request merged**